File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
  def reverseWords(self, s: str) -> str:
    ans = curr = ""
    for ch in s:
      if ch == " ":
        if ans != "" and curr != "":
          ans = curr+" "+ans
        elif curr != "":
          ans = curr
        curr = ""
      else:
        curr += ch
    if ans != "" and curr != "":
      ans = curr+" "+ans
    elif curr != "":
      ans = curr
    return ans

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.ts
function reverseWords(s: string): string {
  let ans = ""
  let curr = ""
  for (let i = 0; i < s.length; i++) {
    if (s[i] === " ") {
      if (ans.length && curr.length) {
        ans = curr + " " + ans
      } else if (curr.length) {
        ans = curr
      }
      curr = ""
    } else {
      curr += s[i]
    }
  }
  if (ans.length && curr.length) {
    ans = curr + " " + ans
  } else if (curr.length) {
    ans = curr
  }
  return ans
}